src/adaptation/stepsize.jl

```diff
@@ -130,10 +130,13 @@ function adapt_stepsize!(
     if !all(isfinite, ϵ)
         @warn "Incorrect ϵ = $ϵ; ϵ_previous = $(da.state.ϵ) is used instead."
         # FIXME: this revert is buggy for batch mode
-        @unpack m, ϵ, x_bar, H_bar = state
+        (; m, ϵ, x_bar, H_bar) = state
     end
 
-    @pack! state = m, ϵ, x_bar, H_bar
+    state.m = m
+    state.ϵ = ϵ
+    state.x_bar = x_bar
+    state.H_bar = H_bar
```

Comment:
Does this change the return value from `state` to `H_bar`?

Reply:
Yes, the `DAState` is a mutable struct, so this is viable, but it’s not good for performance, we need to find a way to avoid such mutable struct if possible.

Reply:
Yeah, this is an inplace function, just return nothing